### PR TITLE
CI: merge sync pr on push to master

### PR DESCRIPTION
**Pull request by maxknv**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [x]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 55d33066845eaf176ce91ead6a36f3cbc93e38c2 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62488/55d33066845eaf176ce91ead6a36f3cbc93e38c2/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/ci/sync_pr.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def main():
+    gh = GitHub(get_best_robot_token())
+
+    pr_info = PRInfo()
+    assert pr_info.merged_pr, "BUG. merged PR number could not been determined"
+
+    prs = gh.get_pulls_from_search(
+        query=f"type:pr [Sync] ClickHouse/ClickHouse#{pr_info.merged_pr}",
```
Comment (Felixoid) [inline tests/ci/sync_pr.py:19]: I think we need to use the query for the `A Sync` URL here. It should work better

File: tests/ci/sync_pr.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    )
+    if len(prs) > 1:
+        print(f"WARNING: More than one PR found [{prs}]")
+    elif len(prs) == 0:
+        print("WARNING: No Sync PR found")
+        sys.exit(0)
+
+    pr = prs[0]
+
+    if pr.state == "closed":
```
Comment (Felixoid) [inline tests/ci/sync_pr.py:30]: These checks for states will go away since the query will cover them.

File: tests/ci/sync_pr.py (hunk 1 above)
Comment (Felixoid) [reply tests/ci/sync_pr.py:19]: `head:{sync_branch_name} org:ClickHouse type:pr state=open`

File: tests/ci/sync_pr.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        repo="ClickHouse/clickhouse-private",
+        #label="pr-sync-upstream",
+    )
+    if len(prs) > 1:
+        print(f"WARNING: More than one PR found [{prs}]")
+    elif len(prs) == 0:
+        print("WARNING: No Sync PR found")
+        sys.exit(0)
+
+    pr = prs[0]
```
[the remark this answers is not in the document]
Comment (maxknv) [reply tests/ci/sync_pr.py:29]: i made it exit with warning if more than one pr found

File: tests/ci/sync_pr.py (hunk 1 above)
Comment (maxknv) [reply tests/ci/sync_pr.py]: done

File: tests/ci/sync_pr.py (hunk 2 above)
Comment (maxknv) [reply tests/ci/sync_pr.py:31]: I added this for better visibility

Comment (Felixoid) [reply tests/ci/sync_pr.py:31]: There should be another filter in the query `state:open`, so we won't get more than one PR

**Pull request merged**